src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/SystemFonts.cs

```diff
@@ -422,15 +422,24 @@ public static TextDecorationCollection StatusFontTextDecorations
         ///     Maps to SPI_NONCLIENTMETRICS
         /// </summary>
         public static double MessageFontSize
+        {
+            get
+            {
+                return ConvertFontHeight(SystemParameters.NonClientMetrics.lfMessageFont.lfHeight);
+            }
+        }
+
+        internal static double ThemeMessageFontSize
         {
             get
             {
                 // TODO : Find a better solution to this. Difference in default size of font in Fluent and other themes.
                 if(ThemeManager.IsFluentThemeEnabled)
                 {
-                    return ThemeManager.DefaultFluentThemeFontSize;
+                    return MessageFontSize * ThemeManager.DefaultFluentFontSizeFactor ;
```

Comment:
Is there a reason we are not doing this in `MessageFontSize` itself? I can think of a case where someone is using the above public property and using Fluent. This will still posses the scaling issues there.

Reply:
Yes because `MessageFontSize` is documented to return `SPI_GETNONCLIENTMETRICS` value.

Reply:
As @miloush says, we've raised this in previous comments. Those properties are representing `SystemParametersInfo`, besides being adjusted for DPI, it is partially a raw abstraction that shouldn't change based on active theme.

A new API (e.g. property) would be suitable if we want to return it publicly imho.

Reply:
On top of that, we should avoid mixing theme-specific code and logic into non-theme assemblies.